### DISCUSSION TITLE
How to convert a 2D ByteArray to bitmap?

### DISCUSSION THREAD

**Discussion by vincentme** [microsoft/onnxruntime#20256] [Mobile Q&A]
This should be a dumb question. Because I am a pure novince to android and kotlin, but now I need to create a demo to run a onnx model. It works on the desktop python api. 

I followed onnxruntime-inference-examples-main/mobile/examples/super_resolution exactly. But the output of my model is a 2D ByteArray (unit8) image. So how to convert it to a bitmap for display? 

This does not work, since the result of  super_resolution was encoded, by mine is not. 

val rawOutput = (output?.get(0)?.value) as ByteArray
val outputImageBitmap =  byteArrayToBitmap(rawOutput)

private fun byteArrayToBitmap(data: ByteArray): Bitmap {
        return BitmapFactory.decodeByteArray(data, 0, data.size)
    }

now the error is `java.lang.ClassCastException: byte[][] cannot be cast to byte[]`

**Comment by Craigacp** [comment]
If you pull it out as a `ByteBuffer` then you might be able to feed it directly into `Bitmap.copyPixelsFromBuffer` assuming that the image is the right orientation and the channels are in the right place. This is an example of unpacking a FloatBuffer into a Java BufferedImage, but I'm not sure how to get from a BufferedImage into an Android Bitmap, and you'll also need to understand the orientation and channels behaviour of your output - https://github.com/oracle/sd4j/blob/main/src/main/java/com/oracle/labs/mlrg/sd4j/VAEDecoder.java#L119.
